### PR TITLE
D2IQ-67977: valuesRemap for auth-url

### PR DESCRIPTION
**Pull request by jr0d**
**What type of PR is this?**

Chore

**What this PR does/ why we need it**:

Adds valuesRemap for auth-url so these addons can be configured by kommander for SSO support

**Which issue(s) this PR fixes**:

https://jira.d2iq.com/browse/D2IQ-67977

**Special notes for your reviewer**:
delta commit: https://github.com/mesosphere/kubernetes-base-addons/pull/242/commits/886fb6bff5ba2ba89ed1cf3ad0718a39b9bc5b01

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Checklist**

* [ ] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.

### CODE REVIEW

File: addons/opsportal/1.1.x/opsportal.yaml [hunk 1]
```diff
@@ -29,6 +29,8 @@ spec:
     chart: opsportal
     repo: https://mesosphere.github.io/charts/stable
     version: 0.3.12
+    valuesRemap:
```
Comment (samvantran) [inline addons/opsportal/1.1.x/opsportal.yaml:32]: how come in this one you renamed the file w/o the revision number but all the other addons maintain them?

Comment (jr0d) [reply addons/opsportal/1.1.x/opsportal.yaml:32]: That is how the addon exists on master: https://github.com/mesosphere/kubernetes-base-addons/tree/master/addons/opsportal/1.1.x

File: addons/dashboard/2.0.x/dashboard-2.yaml [hunk 2]
```diff
@@ -31,6 +31,8 @@ spec:
     chart: kubernetes-dashboard
     repo: https://mesosphere.github.io/charts/staging
     version: 2.0.0
+    valuesRemap:
+      "ingress.annotations.traefik\\.ingress\\.kubernetes\\.io/auth-url": "ingress.auth.auth-url"
```
Comment (joejulian) [inline addons/dashboard/2.0.x/dashboard-2.yaml:35]: Ewww... that's ugly. Can't we do this with out `\\`

Comment (jr0d) [reply addons/dashboard/2.0.x/dashboard-2.yaml:35]: I agree, but we can't unless we change the way valuesRemap works. The helm code (what kubeaddons is using to walk maps) uses "."s to traverse nested maps. It supports escaping `.`s with a `\`. YAML also supports escaping and requires a literal `\\` to be used for key names.

Comment (jr0d) [reply addons/dashboard/2.0.x/dashboard-2.yaml:35]: Hmm.. actually removing the `""` around the key name allows for only one escape. At least with yq.

File: addons/opsportal/1.1.x/opsportal.yaml (hunk 1 above)
Comment (juliangieseke) [reply addons/opsportal/1.1.x/opsportal.yaml:32]: I'll align opsportal with other addons after this release… initial thinking was that this repo does only support snapshots via tags anyway so history is done via git tags, but given its the only addon doing it like that its confusing and should be aligned with other addons.

File: addons/dashboard/2.0.x/dashboard-1.yaml [hunk 3]
```diff
@@ -6,7 +6,7 @@ metadata:
   labels:
     kubeaddons.mesosphere.io/name: dashboard
   annotations:
-    catalog.kubeaddons.mesosphere.io/addon-revision: "2.0.0-beta6"
+    catalog.kubeaddons.mesosphere.io/addon-revision: "2.0.0-1"
```
Comment (makkes) [inline addons/dashboard/2.0.x/dashboard-1.yaml:9]: This sticks out to me, moving from beta6 to 1. Just mentioning here to make sure this is what we want.

**Comment by makkes** [comment]
assigning myself as well so I can track CI checks and merge if they succeed.

**Comment by jr0d** [comment]
closing per: https://github.com/mesosphere/kubernetes-base-addons/pull/315

**Pull request closed without merging**